asciigraf/asciigraf.py

```diff
@@ -15,18 +15,30 @@ def graph_from_ascii(network_string):
         "|":  [Point(0, -1), Point(0, 1)]
     }
     EDGE_CHARS = {"\\", "-", "/", "|"}
-    nodes = OrderedDict(node_iter(network_string))
-
-    node_chars = {}
-    for node_label, pos in nodes.items():
-        for offset, char in enumerate(node_label):
-            node_chars[pos + Point(offset, 0)] = node_label
+    nodes = list(node_iter(network_string))
+
+    node_chars = OrderedDict()
+    line_labels = {}
+    line_label_char_positions = set()
+    for node_label, root_position in nodes:
+        node_label_char_positions = {
+            root_position + Point(offset, 0)
+            for offset, _ in enumerate(node_label)
+        }
+        if node_label.startswith("(") and node_label.endswith(")"):
+            label_value = node_label[1:-1]
+            line_labels[root_position] = label_value
+            line_label_char_positions |= node_label_char_positions
+        else:
+            node_chars.update(
+                (pos, node_label) for pos in node_label_char_positions
+            )
 
-    # nodes = node_chars
+    # we'll treat edge labels (e.g. "(my_label)") as consecutive "-" edge chars
```

Comment:
I don't agree because renaming it means we changed the length, though I understand it's probably easier for other reasons like it makes it simpler to align with other lines and points on the graph.

Reply:
So basically I agree but I don't 100% love it. It's pragmatic and simple which is good.

Reply:
We could go back at the end and modify the lengths of all the edges by subtracting the length of the label.